subprojects/docs/src/samples/groovy/compilationAvoidance/groovy/build.gradle

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+subprojects {
+    apply plugin: 'groovy'
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        implementation localGroovy()
+    }
+}
+
+project(':astTransformationConsumer') {
+// tag::groovy-compilation-avoidance[]
+    configurations { astTransformation }
```

Comment:
I try to reduce the noise (line jumping) when switching between Kotlin and Groovy samples. So I would put the `astTransformation` on a new line here or use `configurations.create` in the Kotlin version. You could even use `configurations.create` in both versions.

Reply:
Here's that part of the sample in Kotlin then Groovy with the same lines count and idiomatic DSL:

Kotlin
```kotlin
project("astTransformationConsumer") {
    val astTransformation by configurations.creating
    dependencies {
        astTransformation(project(":astTransformation"))
    }
    tasks.withType<GroovyCompile> {
        compilerPluginClasspath.from(astTransformation)
    }
}
```

Groovy
```groovy
project("astTransformationConsumer") {
    configurations { astTransformation }
    dependencies {
        astTransformation(project(":astTransformation"))
    }
    tasks.withType<GroovyCompile> {
        compilerPluginClasspath.from(configurations.astTransformation)
    }
}
```

Reply:
Put on a new line

```
configurations { 
    astTransformation 
}
```

Reply:
@blindpirate Please don't use the new line so both Kotlin and Groovy samples have the same number of lines.

Reply:
Well, where does the restriction of `Groovy Kotlin samples must have same number of lines` come from?

Reply:
I wouldn't say it's a restriction. But we should try to make them same length when it is possible in order for the page content not to jump when switching from one DSL to another in the browser. It is very much possible with this sample, so better do it.

Reply:
Understood. Thanks for the explanation @eskatos !